extension.js

```diff
@@ -168,6 +179,48 @@ const ExtensionLayout = new Lang.Class({
     GLib.spawn_command_line_async(cmd);
   },
 
+  _findNewStreamerEntries:function(lastList, currentList, detectGameChange) {
+    detectGameChange = detectGameChange || false;
+    if ( lastList.length == 0 )
+      return currentList;
+
+    let newEntries = [];
+    // filter out all lastList items from currentList
+    currentList.forEach(entry => {
+      let match = false;
+
+      // Use a for loop so we can break it off when we match
+      for ( var i = 0; i < lastList.length; i++ ) {
+        let last = lastList[i];
+        if ( entry.streamer == last.streamer )
+          if ( !detectGameChange || ( detectGameChange && entry.game == last.game ) ) {
+            match = true;
+            break;
+          }
+      }
+
+      if ( !match )
+        newEntries.push(entry);
+    });
+
+    return newEntries;
+  },
+
+  _streamerOnlineNotification:function(streamer) {
+    let notification = new MessageTray.Notification(
+      this.notification_source,
+      streamer.streamer + " is live!",
+      "Playing " + streamer.game);
```

Comment:
Gettext should be as available here as anywhere else. We can just use the underscore

Reply:
I did make an attempt at using the underscore. Step 2 of that was to update the `twitchlive.pot`. I *couldn't figure out* how to generate the catalogue (with `xgettext`), so I didn't include the underscore at all.